docs/specifications/tests/Zone-TP/zone11.md

```diff
@@ -0,0 +1,284 @@
+* [Summary](#summary)
+* [Test procedure](#test-procedure)
+* [Outcome(s)](#outcomes)
+* [Special procedural requirements](#special-procedural-requirements)
+* [Intercase dependencies](#intercase-dependencies)
+* [Terminology](#terminology)
+
+## Objective
+
+Sender Policy Framework (SPF), described in [RFC 7208][RFC7208], is a mechanism
+allowing domain name owners to specify which hosts are allowed to send mail
+claiming to be from that domain. It is implemented by means of TXT records in a
+structured format.
+
+This test case looks up SPF records in the domain to be tested. It checks that
+there is at most one published SPF version 1 policy and, if present, also checks
+the following:
+
+ * the policy has correct syntax;
+ * the policy uses no deprecated [terms][SPF term] nor deprecated [macros][macro].
+
+## Caveat
+
+In order to avoid any denial-of-service attacks, SPF imposes limits on the
+number of recursive DNS lookups an SPF validator may carry out when evaluating a
+policy. This test case will generate an error if it can be certain that a policy
+would generate more than 10 recursive DNS queries by a validator. A policy may
+also use `include` mechanisms or `redirect` modifiers for indirection, but in
+order to keep the test case simple, Zonemaster will not follow these terms.
+
+For that reason, if the published SPF policy contains `include` mechanisms or a
+`redirect` modifier, Zonemaster not raising an error about a too complex policy
+does not mean that the policy will never exceed DNS processing limits specified
+in [RFC 7208, § 4.6.4][RFC7208#4.6.4]. Due to the macro expansion facilities of
+SPF, such problems may only be restricted to certain sender domains.
+
+A true SPF validator would need, as inputs, the IPv4 or IPv6 address of the
+sending mail host along with the domain to test. Consequently, Zonemaster will
+not attempt to evaluate the published SPF policy.
+
+## Scope
+
+It is assumed that *Child Zone* has been tested and reported by
+[Connectivity01]. This test case will just ignore non-responsive name servers or
+name servers not giving a correct DNS response for an authoritative name server.
+
+## Inputs
+
+* "Child Zone" - The domain name to be tested.
+
+## Summary
+
+Message Tag                     | Level   | Arguments                     | Message ID for message tag
+:-------------------------------|:--------|:------------------------------|:--------------------------------------------
+Z11_INCONSISTENT_SPF_POLICIES   | WARNING |                               | The *Child Zone* publishes different SPF policies on two different name servers.
+Z11_NO_SPF_FOUND                | DEBUG   |                               | The *Child Zone* does not publish an SPF policy.
+Z11_SPF1_HAS_P_MACRO            | WARNING | ns_ip_list, spf_policy        | The following SPF version 1 policy contains a %{p} macro, which should no longer be used: {spf_policy}. Policy retrieved from the following nameservers: {ns_ip_list}.
```

Comment:
Based on what I see on the `.fr` zone, 92 % of SPF rules are 80 octets long or less. The 95th percentile is at 93 octets. However, the outliers can be massive, with the largest being 2 132 octets.

While most SPF rules could be expressed literally, maybe the sheer fact that such outliers exist does sort of weigh against including the whole policy in the message. Maybe just leave it out, after all?

Reply:
I suggest to leave it out. Even with a 40 characters policy it will be hard to read the message with first some text and then the policy and then some more text and then a list of IP addresses. I always put the list of IP addresses at the end of the message to make sure it does not prevent the reader from finding the text. (And I have created issue https://github.com/zonemaster/zonemaster-gui/issues/315 for improved readability.)

Reply:
Okay, good point. I’ve removed the placeholders for the SPF policy in my latest iteration.